File: Java/cajero.java
import java.util.Scanner;

public class cajero {
    public static void main(String[] args) {
        double saldo = 1000.0;
        Scanner sc = new Scanner(System.in);
        while (true) {
            System.out.println("\n--- Cajero Automático ---");
            System.out.println("1. Consultar saldo");
            System.out.println("2. Retirar dinero");
            System.out.println("3. Depositar dinero");
            System.out.println("4. Salir");
            System.out.print("Elige una opción: ");
            String opcion = sc.nextLine();
            
            if (opcion.equals("1")) {
                System.out.println("Tu saldo actual es: $" + saldo);
            } else if (opcion.equals("2")) {
                System.out.print("Ingresa la cantidad a retirar: $");
                double cantidad = sc.nextDouble();
                sc.nextLine();
                if (cantidad > saldo) {
                    System.out.println("Fondos insuficientes.");
                } else {
                    saldo -= cantidad;
                    System.out.println("Retiro exitoso. Tu nuevo saldo es: $" + saldo);
                }
            } else if (opcion.equals("3")) {
                System.out.print("Ingresa la cantidad a depositar: $");
                double cantidad = sc.nextDouble();
                sc.nextLine();
                saldo += cantidad;
                System.out.println("Depósito exitoso. Tu nuevo saldo es: $" + saldo);
            } else if (opcion.equals("4")) {
                System.out.println("Gracias por usar el cajero. ¡Hasta luego!");
                break;
            } else {
                System.out.println("Opción inválida, intenta de nuevo.");
            }
        }
        sc.close();
    }
}

File: C++/cajero.cpp
#include <iostream>
#include <string>
using namespace std;

int main() {
    double saldo = 1000.0;
    while (true) {
        cout << "\n--- Cajero Automático ---" << endl;
        cout << "1. Consultar saldo" << endl;
        cout << "2. Retirar dinero" << endl;
        cout << "3. Depositar dinero" << endl;
        cout << "4. Salir" << endl;
        cout << "Elige una opción: ";
        string opcion;
        getline(cin, opcion);

        if (opcion == "1") {
            cout << "Tu saldo actual es: $" << saldo << endl;
        } else if (opcion == "2") {
            cout << "Ingresa la cantidad a retirar: $";
            double cantidad;
            cin >> cantidad;
            cin.ignore();
            if (cantidad > saldo)
                cout << "Fondos insuficientes." << endl;
            else {
                saldo -= cantidad;
                cout << "Retiro exitoso. Tu nuevo saldo es: $" << saldo << endl;
            }
        } else if (opcion == "3") {
            cout << "Ingresa la cantidad a depositar: $";
            double cantidad;
            cin >> cantidad;
            cin.ignore();
            saldo += cantidad;
            cout << "Depósito exitoso. Tu nuevo saldo es: $" << saldo << endl;
        } else if (opcion == "4") {
            cout << "Gracias por usar el cajero. ¡Hasta luego!" << endl;
            break;
        } else {
            cout << "Opción inválida, intenta de nuevo." << endl;
        }
    }
    return 0;
}
